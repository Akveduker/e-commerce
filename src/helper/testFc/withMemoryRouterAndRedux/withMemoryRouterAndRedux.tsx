import { configureStore } from "@reduxjs/toolkit"
import { render } from "@testing-library/react"
import { InitialEntry } from "history"
import { Provider } from "react-redux"
import { MemoryRouter } from "react-router-dom"
import { mockReducersForStore } from "../../mock/mockStore/mockStore"
export const withMemoryRouterAndRedux = (children: React.ReactNode, reducers?: Partial<typeof mockReducersForStore>, initialEntries?: InitialEntry[]) => {
    const mockStore = configureStore({
        reducer: { ...mockReducersForStore, ...reducers }
    });
    return render(
        <MemoryRouter initialEntries={initialEntries}>
            <Provider store={mockStore}>

                {children}
            </Provider>
        </MemoryRouter>
    )
}