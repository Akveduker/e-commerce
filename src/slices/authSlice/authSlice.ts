import { storageNames } from './../../data/storage/storageNames';
import { IAuthState } from '../../types/Authorization/Authorization';
import { createSlice } from '@reduxjs/toolkit';
const storage = localStorage.getItem(storageNames.auth)
let initialState: IAuthState = {
    accessToken: '',
    id: 0,
    expiresAt: 0,
}
if (storage) {
    const auth: IAuthState = JSON.parse(storage)
    if (Date.now() >= auth.expiresAt) {
        localStorage.removeItem(storageNames.auth)
    } else {
        initialState = auth
    }

}

export const createAuthSlice = (newState = initialState) => {
    const slice = createSlice({
        name: 'token',
        initialState: newState,
        reducers: {
            setToken: (state, { payload }: { payload: string }) => {
                state.accessToken = payload
            },
            setFullAuthState: (state, { payload }: { payload: IAuthState }) => {
                state.accessToken = payload.accessToken
                state.id = payload.id
            },
            logoutAuth: (state) => {
                localStorage.removeItem(storageNames.auth)
                return { accessToken: '', id: 0, expiresAt: 0 }
            }
        }
    })
    return slice
}
const authSlice = createAuthSlice()
export const { setToken, setFullAuthState, logoutAuth } = authSlice.actions
export const authReducer = authSlice.reducer