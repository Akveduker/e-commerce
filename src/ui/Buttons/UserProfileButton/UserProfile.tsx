import { FC } from 'react';
import { Link } from 'react-router-dom';
import { useGetAuthRoot } from '../../../hooks/roots/useGetAuthRoot';
import UserProfileIcon from '../../icons/UserProfileIcon/UserProfileIcon';
import s from './UserProfile.module.scss'
const UserProfile: FC = () => {
    const root = useGetAuthRoot()
    return (
        <Link to={root} className={s['link']}>
            <UserProfileIcon />
        </Link>
    );
};

export default UserProfile;