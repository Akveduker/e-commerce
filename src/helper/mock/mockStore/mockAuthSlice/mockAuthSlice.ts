
import { IAuthState } from "../../../../types/Authorization/Authorization"
import { mockObject } from "../../mockObject/mockObject"
import { createAuthSlice } from '../../../../slices/authSlice/authSlice';

export const mockAuthInitial: IAuthState = {
    accessToken: 'token',
    id: 1,
    expiresAt: 232222222,
}
export const mockAuthSlice = (item?: Partial<IAuthState>) => mockObject(mockAuthInitial, item)
export const mockAuthReducer = createAuthSlice(mockAuthInitial).reducer