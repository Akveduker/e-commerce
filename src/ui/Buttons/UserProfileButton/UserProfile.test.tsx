import { mockAuthSlice } from "../../../helper/mock/mockStore/mockAuthSlice/mockAuthSlice"
import { withMemoryRouterAndRedux } from "../../../helper/testFc/withMemoryRouterAndRedux/withMemoryRouterAndRedux"
import { createAuthSlice } from "../../../slices/authSlice/authSlice"
import UserProfile from "./UserProfile"

describe('UserProfile', () => {
    test('Снапшот с рутом на авторизацию', () => {
        const authReducerNotLoggedIn = createAuthSlice(mockAuthSlice({ accessToken: '', id: 0 })).reducer
        const { container } = withMemoryRouterAndRedux(
            <UserProfile />,
            {
                auth: authReducerNotLoggedIn
            }
        )
        expect(container).toMatchSnapshot()
    })
    test('Снапшот с рутом на профиль', () => {
        const { container } = withMemoryRouterAndRedux(
            <UserProfile />
        )
        expect(container).toMatchSnapshot()
    })
})