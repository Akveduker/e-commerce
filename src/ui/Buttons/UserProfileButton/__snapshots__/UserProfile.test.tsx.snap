// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile Снапшот с рутом на авторизацию 1`] = `
<div>
  <a
    class="link"
    href="/authorization"
  >
    <svg
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 21.9999L3.79 19.1199C6.4 9.6199 17.6 9.6199 20.21 19.1199L21 21.9999"
        stroke="#151515"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2.5"
      />
      <path
        d="M12 11.98C14.7614 11.98 17 9.74141 17 6.97998C17 4.21856 14.7614 1.97998 12 1.97998C9.23858 1.97998 7 4.21856 7 6.97998C7 9.74141 9.23858 11.98 12 11.98Z"
        stroke="#151515"
        stroke-linecap="round"
        stroke-linejoin="bevel"
        stroke-width="2.5"
      />
    </svg>
  </a>
</div>
`;

exports[`UserProfile Снапшот с рутом на профиль 1`] = `
<div>
  <a
    class="link"
    href="/user/profile"
  >
    <svg
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 21.9999L3.79 19.1199C6.4 9.6199 17.6 9.6199 20.21 19.1199L21 21.9999"
        stroke="#151515"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2.5"
      />
      <path
        d="M12 11.98C14.7614 11.98 17 9.74141 17 6.97998C17 4.21856 14.7614 1.97998 12 1.97998C9.23858 1.97998 7 4.21856 7 6.97998C7 9.74141 9.23858 11.98 12 11.98Z"
        stroke="#151515"
        stroke-linecap="round"
        stroke-linejoin="bevel"
        stroke-width="2.5"
      />
    </svg>
  </a>
</div>
`;
